ci-setups/mr-title-validation-gitlab.md

```diff
@@ -0,0 +1,114 @@
+
+# Merge Request Title Validation CI/CD Setup on GitLab
+
+## Table of Contents
+
+- [1. Introduction](#1-introduction)
+- [2. Getting Started](#2-getting-started)
+- [3. Workflow Configuration](#3-workflow-configuration)
+  - [3.1 Stages](#31-stages)
+  - [3.2 Tags](#32-tags)
+  - [3.3 Script](#33-script)
+- [4. Example YAML Configuration](#4-example-yaml-configuration)
+- [5. Validation Checks](#5-validation-checks)
+- [6. Troubleshooting](#6-troubleshooting)
+- [7. Sample Repository](#7-sample-repository)
+- [8. Conclusion](#8-conclusion)
+  - [Benefits of CI Setup](#benefits-of-ci-setup)
+  - [Future Enhancements](#future-enhancements)
+
+## 1. Introduction
+
+This document provides guidelines for setting up Continuous Integration and Continuous Deployment (CI/CD) to validate MR Title for projects on GitLab. It aims to ensure consistent merge request titles across all projects.
+
+[Back to top](#table-of-contents)
+
+## 2. Getting Started
+
+Before you begin, ensure you have a Gitlab account, and a basic understanding of Gitlab workflows.
+
+[Back to top](#table-of-contents)
+
+## 3. Workflow Configuration
+
+Create a `.gitlab-ci.yml` file in your project to define your workflow.
+
+### 3.1 Stages
+
+Specify in which stage your workflow should run. Stage used is `validate`.
+
+### 3.2 Tags
+
+Define the gitlab runner tag in which your workflow will run. For example - `python-3.11.9`.
+
+### 3.3 Script
+
+Specify the logic of the checks which needs to be checked for merge request validation.
+
+[Back to top](#table-of-contents)
+
+## 4. Example YAML Configuration
+
+Here is an example of a basic Gitlab Actions workflow file for MR Title Validation:
+
+```yaml
+stages:
+  - validate
+
+validate_merge_request_title:
+  stage: validate
+  tags:
+    - python-3.11.9
+  script:
+    - 'echo "Validating merge request title: $CI_MERGE_REQUEST_TITLE"'
+    - |
+      if echo "$CI_MERGE_REQUEST_TITLE" | grep -Eq "^(build|chore|ci|docs|feat|fix|perf|refactor|style|test|sample): [a-zA-Z0-9 ]{0,50}$"; then
```

Comment:
Similar to https://github.com/OsmosysSoftware/dev-standards/pull/69, let's only allow lowercase letters in the PR titles.

Reply:
Done, updated the code and the document.